maintenance/liftOver_assemblies.py

```diff
@@ -0,0 +1,540 @@
+    ax0.legend(title=chromName)
+    ax0.set_xlabel("Chromosome position")
+    ax0.set_ylabel("Rate deviation (cM/Mb)")
+    ax1 = plt.subplot(gs[1])
+    bp = ax1.violinplot(D, showmeans=True, widths=0.95)
+    ax2 = plt.subplot(gs[2])
+    bp = ax2.boxplot(D, notch=True, showfliers=False)
+    plt.savefig(pdf_file)
+
+
+def allDev_plot(dev, validationDir, genome):
+    """
+    plots the deviation between rates for the original and the re-lifted original
+    for all chromosomes as a single boxplot
+    """
+    rates, labels = [], []
+    for chrom in genome.chromosomes:
+        for d in dev:
+            if d[0][3:] == chrom.id:
+                rates.append(d[1])
+                labels.append(d[0])
+    pdf_file = os.path.join(validationDir, "all_chromosome_deviations.pdf")
+    plt.figure(figsize=(10,4))
+    plt.xlabel("Chromosome")
+    plt.ylabel("Rate deviation (cM/Mb)")
+    plt.xticks(rotation=45)
+    plt.boxplot(rates, labels=labels, notch=True, showfliers=False)
+    plt.savefig(pdf_file)
+
+
+def worker(task_q, result_q, params):
+    """
+    does the work
+    """
+    intermediatesDir, validationDir, inFiles, outFiles, chainFile, winLen, gapThresh, validation, adjAvg = params
+    while True:
+        try:
+            index_range, nth_job = task_q.get()
+            tmp = []
+            for i in index_range:
+                # convert hapmap to bed
+                tmp1, chromName, oldRates, weights, oldRateDict, old_average = hapmap_to_bed(inFiles[i], outFiles[i])
+                if chromName == None:
+                    break
+                # liftOver files
+                tmp2, tmp2unmapped = lift_over(tmp1, outFiles[i], chainFile)
+                # convert bed back to hapmap
+                joinedRates, new_average = bed_to_hapmap(tmp2, outFiles[i], chromName, winLen, gapThresh, i, oldRateDict, adjAvg)
+                # move intermediates to new directory
+                cmd = [str(x) for x in ["mv",tmp1,tmp2,tmp2unmapped,intermediatesDir]]
+                sp.run(cmd)
+                # plot the rates
+                if validation:
+                    tmp1, chromName, oldRates, weights, oldRateDict, old_average = hapmap_to_bed(validation[i], outFiles[i])
+                    win_pos, D = get_deviation(oldRates, weights, joinedRates, winLen, i, new_average)
+                    deviation_plot(validationDir, win_pos, D, chromName, new_average)
+                    tmp.append([chromName,D])
+                else:
+                    step_plot(validationDir, oldRates, joinedRates, chromName, old_average, new_average)
+            if validation:
+                result_q.put(tmp)
+        finally:
+            task_q.task_done()
+
+
+def remap(intermediatesDir, validationDir, inFiles, outFiles, chainFile, winLen, gapThresh, adjAvg, nProc, validation=False, genome=False):
+    """
+    run all functions to translate and liftOver file types
+    """
+    # multithread remapping
+    task_q = mp.JoinableQueue()
```

Comment:
This does seem unneccesarily low-level here - there's a good example in the [msprime code](https://github.com/tskit-dev/msprime/blob/main/verification.py#L4755) of using concurrent.futures along with a tqdm progress bar.